pkg/operator/client/client.go

```diff
@@ -0,0 +1,61 @@
+package client
+
+import (
+	"fmt"
+
+	configv1 "github.com/openshift/api/config/v1"
+	operatorv1 "github.com/openshift/api/operator/v1"
+
+	kscheme "k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+var (
+	// scheme contains all the API types necessary for the operator's dynamic
+	// clients to work. Any new non-core types must be added here.
+	//
+	// NOTE: The discovery mechanism used by the client won't automatically refresh,
+	// so only add types here that are _guaranteed_ to exist before the operator
+	// starts.
+	scheme *runtime.Scheme
+)
+
+func init() {
+	scheme = kscheme.Scheme
+	if err := operatorv1.AddToScheme(scheme); err != nil {
+		panic(err)
+	}
+	if err := configv1.Install(scheme); err != nil {
+		panic(err)
+	}
+}
+
+func GetScheme() *runtime.Scheme {
+	return scheme
+}
+
+// NewClient builds an operator-compatible kube client from the given REST config.
+func NewClient(kubeConfig *rest.Config) (client.Client, error) {
+	managerOptions := manager.Options{
+		Scheme:         scheme,
+		MapperProvider: apiutil.NewDiscoveryRESTMapper,
+	}
+	mapper, err := managerOptions.MapperProvider(kubeConfig)
```

Comment:
This is just a suggestion for a follow-up: Do we need `managerOptions`, or would the following work?
```go
	mapper, err := apiutil.NewDiscoveryRESTMapper(kubeConfig)
```

Reply:
done